File: tests/part_xml_test.php
<?php
/*
 * OpenCReports test
 * Copyright (C) 2019-2024 Zoltán Böszörményi <zboszor@gmail.com>
 * See COPYING.LGPLv3 in the toplevel directory.
 */

require_once 'test_common.php';

$coltypes = [
	OpenCReport::RESULT_NUMBER, OpenCReport::RESULT_STRING, OpenCReport::RESULT_STRING, OpenCReport::RESULT_NUMBER, OpenCReport::RESULT_NUMBER
];

$o = new OpenCReport();

if (!$o->parse_xml("part_xml_test.xml")) {
	echo "XML parse error" . PHP_EOL;
	exit(0);
}

for ($p = $o->part_get_first(), $i = 0; isset($p); $p = $p->get_next(), $i++) {
	$partname = "part " . $i;
	print_part_reports($partname, $p);
}

File: tests/part_xml_test.c
/*
 * OpenCReports test
 * Copyright (C) 2019-2025 Zoltán Böszörményi <zboszor@gmail.com>
 * See COPYING.LGPLv3 in the toplevel directory.
 */

#include <stdio.h>

#include <opencreport.h>
#include "test_common.h"

const int32_t coltypes[5] = {
	OCRPT_RESULT_NUMBER, OCRPT_RESULT_STRING, OCRPT_RESULT_STRING, OCRPT_RESULT_NUMBER, OCRPT_RESULT_NUMBER
};

int main(int argc, char **argv) {
	opencreport *o = ocrpt_init();
	ocrpt_list *pl = NULL;
	ocrpt_part *p;
	int32_t i;

	if (!ocrpt_parse_xml(o, "part_xml_test.xml")) {
		printf("XML parse error\n");
		ocrpt_free(o);
		return 0;
	}

	for (p = ocrpt_part_get_next(o, &pl), i = 0; p; p = ocrpt_part_get_next(o, &pl), i++) {
		char partname[16];

		sprintf(partname, "part %d", i);
		print_part_reports(partname, p);
	}

	ocrpt_free(o);

	return 0;
}
